pkg/awsManager/ec2Manager.go

```diff
@@ -60,18 +62,20 @@ func DeleteEc2Instance(client clientpkg.Client, EC2InstancesToBeDeleted []*strin
 		return nil
 	}
 
-	_, err := client.TerminateInstances(&ec2.TerminateInstancesInput{InstanceIds: EC2InstancesToBeDeleted})
-	if err != nil {
-		if err, ok := err.(awserr.Error); ok {
-			switch err.Code() {
-			default:
-				logger.Error(err, "Failed to delete instances provided", "Instances", &EC2InstancesToBeDeleted)
-			}
-		} else {
-			logger.Error(err, "Failed to delete instances provided", "Instances", &EC2InstancesToBeDeleted)
+	// We're batching the deletes to avoid hitting the limit of instances per request
+	totalToDelete := len(EC2InstancesToBeDeleted)
+	for lowerBound, upperBound := 0, 0; lowerBound <= totalToDelete-1; lowerBound = upperBound {
+		upperBound = lowerBound + maxBatchSize
+		if upperBound > totalToDelete {
+			upperBound = totalToDelete
+		}
+		batchedEC2Instances := EC2InstancesToBeDeleted[lowerBound:upperBound]
+		_, err := client.TerminateInstances(&ec2.TerminateInstancesInput{InstanceIds: batchedEC2Instances})
+		if err != nil {
+			localMetrics.ResourceFail(localMetrics.Ec2Instance, client.GetRegion())
+			logger.Error(err, "Failed to delete instances provided", "Instances", &batchedEC2Instances)
+			return err
```

Comment:
This looks like a very smart sliding window function, but trying to understand it made me a bit of a headache ;)
I wonder if there is an easier-to-read way of implementing this, e.g. by popping the deleted instances out of  `EC2InstancesToBeDeleted` slice in each iteration.

Reply:
While I agree with the readability thing, I think just adding a comment here to explain what this is doing can unblock this PR, and then if we find later that this is hard to read or understand we can make changes.  Especially since we have unit tests here, refactoring this later should be relatively safe to do.